test/project_types/php/commands/serve_test.rb

```diff
@@ -104,17 +113,56 @@ def test_server_command_with_invalid_host_url
         ).never
 
         assert_raises ShopifyCLI::Abort do
-          run_cmd("app php serve")
+          run_cmd("app php serve --host=#{invalid_host}")
         end
       end
 
-      def test_update_env_with_host
-        ShopifyCLI::Tunnel.expects(:start).never
+      def test_server_command_when_port_passed
+        ShopifyCLI::Tunnel.stubs(:start).returns("https://example.com")
         ShopifyCLI::Tasks::UpdateDashboardURLS.expects(:call)
-        ShopifyCLI::Resources::EnvFile.any_instance.expects(:update).with(
-          @context, :host, "https://example-foo.com"
+        ShopifyCLI::Resources::EnvFile.any_instance.expects(:update)
+        ShopifyCLI::ProcessSupervision.expects(:running?).with(:npm_watch).returns(false)
+        ShopifyCLI::ProcessSupervision.expects(:stop).never
+        ShopifyCLI::ProcessSupervision.expects(:start).with(:npm_watch, "npm run watch", force_spawn: true)
+
+        @context.expects(:system).with(
+          "php",
+          "artisan",
+          "serve",
+          "--port",
+          "5000",
+          env: {
+            "SHOPIFY_API_KEY" => "mykey",
+            "SHOPIFY_API_SECRET" => "mysecretkey",
+            "SHOP" => "my-test-shop.myshopify.com",
+            "SCOPES" => "read_products",
+            "HOST" => "https://example.com",
+            "DB_DATABASE" => "storage/db.sqlite",
+          }
         )
-        run_cmd('app php serve --host="https://example-foo.com"')
+
+        @context.expects(:puts).with(
+          "\n" +
+          @context.message("php.serve.open_info", "https://example.com/login?shop=my-test-shop.myshopify.com") +
+          "\n"
+        )
+
+        run_cmd("app php serve --port=5000")
+      end
+
+      def test_server_command_when_invalid_port_passed
+        invalid_port = "NOT_PORT"
```

Comment:
I updated the invalid host unit test to take in the invalid host url directly from the command instead of setting Tunnel.start (ngrok) since Tunnel.start would likely always return a valid host url. This allows the tests to pass with the new changes. @pepibumur @paulomarg

Reply:
That's much better. Thanks for changing that @molly-yu